src/CodeExplorerBundle/Twig/SourceCodeExtension.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace CodeExplorerBundle\Twig;
+
+/**
+ * CAUTION: this is an extremely advanced Twig extension. It's used to get the
+ * source code of the controller and the template used to render the current
+ * page. If you are starting with Symfony, don't look at this code and consider
+ * studying instead the code of the src/AppBundle/Twig/AppExtension.php extension.
+ *
+ * @author Ryan Weaver <weaverryan@gmail.com>
+ * @author Javier Eguiluz <javier.eguiluz@gmail.com>
+ */
+class SourceCodeExtension extends \Twig_Extension
+{
+    private $controller;
+    private $kernelRootDir;
+
+    public function __construct($kernelRootDir)
+    {
+        $this->kernelRootDir = $kernelRootDir;
+    }
+
+    public function setController($controller)
+    {
+        $this->controller = $controller;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getFunctions()
+    {
+        return array(
+            new \Twig_SimpleFunction('show_source_code', array($this, 'showSourceCode'), array('is_safe' => array('html'), 'needs_environment' => true)),
+        );
+    }
+
+    public function showSourceCode(\Twig_Environment $twig, \Twig_Template $template)
+    {
+        return $twig->render('@CodeExplorer/source_code.html.twig', array(
+            'controller' => $this->getController(),
+            'template'   => $this->getTemplateSource($template),
+        ));
+    }
+
+    private function getController()
+    {
+        // this happens for example for exceptions (404 errors, etc.)
+        if (null === $this->controller) {
+            return;
+        }
+
+        $method = $this->getCallableReflector($this->controller);
+
+        $classCode = file($method->getFilename());
+        $methodCode = array_slice($classCode, $method->getStartline() - 1, $method->getEndLine() - $method->getStartline() + 1);
+        $controllerCode = '    '.$method->getDocComment()."\n".implode('', $methodCode);
+
+        return array(
```

Comment:
__[development][code_sniffer]__
* Add a comma after each item in a multi-line array

<!-- Comment by [SideCI](https://sideci.com) -->